parking_lot/parking_lot.py

```diff
@@ -1,8 +1,150 @@
 # -*- coding: utf-8 -*-
-from flask import Flask
+import datetime
 
-app = Flask(__name__)
 
-@app.route("/")
-def hello_world():
-    return "<p>Hello, World!</p>"
+class Parking_lot_error( Exception ):
+    message = 'unhandle error with parking lot'
+
+    def __init__( self, message=None ):
+        if message is None:
+            message = self.message
+        self.data = {
+            'status': 'error',
+            'error': message
+        }
+
+
+class Full_parking_error( Parking_lot_error ):
+    message = 'No free space.'
+
+
+class Cannot_find_car_error( Parking_lot_error ):
+    message = 'Cannot find a car in the location {}.'
+
+    def __init__( self, location=None ):
+        message = self.message.format( location )
+        self.data = {
+            'status': 'error',
+            'error': message
+        }
+
+
+class Car:
```

Comment:
this class feels more like a ticket because it have entry and exit timestamps, license_plate of the car, tariff, the metadata needed for calculating fees on car's exit.

Reply:
yes, you are right, ticket can be a better metaphor, i was thinking in car inside of the parking lot.